From 3c91458ecf0da7b820f6c31efab2cf1e569ae146 Mon Sep 17 00:00:00 2001
From: Chirayu Desai <chirayudesai1@gmail.com>
Date: Tue, 7 Mar 2023 01:55:33 +0530
Subject: [PATCH] try policy

* https://developer.android.com/reference/android/os/Bundle#putShort(java.lang.String,%20short)
* "Inserts a short value into the mapping of this Bundle, replacing any existing value for the given key."
---
 .../policy/AppRestrictionsProvider.java       | 30 +++++++++++++++++--
 1 file changed, 28 insertions(+), 2 deletions(-)

diff --git a/components/policy/android/java/src/org/chromium/components/policy/AppRestrictionsProvider.java b/components/policy/android/java/src/org/chromium/components/policy/AppRestrictionsProvider.java
index 85cf94e4e6ec9..4dfc525a5ce1d 100644
--- a/components/policy/android/java/src/org/chromium/components/policy/AppRestrictionsProvider.java
+++ b/components/policy/android/java/src/org/chromium/components/policy/AppRestrictionsProvider.java
@@ -24,12 +24,38 @@ public class AppRestrictionsProvider extends AbstractAppRestrictionsProvider {
      */
     public static Bundle getApplicationRestrictionsFromUserManager(
             UserManager userManager, String packageName) {
+        Bundle restrictions = new Bundle();
         try {
-            return userManager.getApplicationRestrictions(packageName);
+            restrictions = userManager.getApplicationRestrictions(packageName);
         } catch (SecurityException e) {
             // Android bug may throw SecurityException. See crbug.com/886814.
-            return new Bundle();
+            // Do nothing
         }
+        // https://chromeenterprise.google/policies/#BrowserSignin
+        restrictions.putInt("BrowserSignin", 0);
+        // https://chromeenterprise.google/policies/#ComponentUpdatesEnabled
+        restrictions.putBoolean("ComponentUpdatesEnabled", false);
+        // https://chromeenterprise.google/policies/#ContextualSearchEnabled
+        restrictions.putBoolean("ContextualSearchEnabled", false);
+	// https://chromeenterprise.google/policies/#DomainReliabilityAllowed
+        restrictions.putBoolean("DomainReliabilityAllowed", false);
+        // https://chromeenterprise.google/policies/#MetricsReportingEnabled
+        restrictions.putBoolean("MetricsReportingEnabled", false);
+	// https://chromeenterprise.google/policies/#NTPContentSuggestionsEnabled
+	restrictions.putBoolean("NTPContentSuggestionsEnabled", false);
+        // https://chromeenterprise.google/policies/#NetworkPredictionOptions
+        restrictions.putInt("NetworkPredictionOptions", 2);
+        // https://chromeenterprise.google/policies/#SearchSuggestEnabled
+        restrictions.putBoolean("SearchSuggestEnabled", false);
+        // https://chromeenterprise.google/policies/#PrivacySandbox
+        restrictions.putBoolean("PrivacySandboxAdMeasurementEnabled", false);
+        restrictions.putBoolean("PrivacySandboxAdTopicsEnabled", false);
+        restrictions.putBoolean("PrivacySandboxPromptEnabled", false);
+        restrictions.putBoolean("PrivacySandboxSiteEnabledAdsEnabled", false);
+        // https://chromeenterprise.google/policies/#SafeBrowsing
+        restrictions.putBoolean("SafeBrowsingExtendedReportingEnabled", false);
+        restrictions.putInt("SafeBrowsingProtectionLevel", 0);
+        return restrictions;
     }
 
     private final UserManager mUserManager;
-- 
2.39.2

